### PR TITLE
✨ Expose Approximation Methods in Python

### PR DESCRIPTION
**Pull request by burgholzer**
This PR adds the necessary changes to expose the existing approximation functionality to Python.
Changes are really minor; mostly unification of the simulator constructors.

While writing tests, I even found and subsequently fixed a bug in the implementation that prevented nodes at level 0 from being collected.

### CODE REVIEW

File: include/CircuitSimulator.hpp [hunk 1]
```diff
@@ -12,60 +12,61 @@
 #include <string>
 
 struct ApproximationInfo {
-    enum ApproximationWhen {
+    enum ApproximationMethod {
```
Comment (hillmich) [inline include/CircuitSimulator.hpp:15]: `ApproximationMethod` seems like a misleading name (especially if we want to introduce removing edges to the approximation. How about `ApproximationDriver` oder `ApproximationStrategy` (in analogy to the terms in the paper)?

File: src/Simulator.cpp [hunk 2]
```diff
@@ -178,9 +178,16 @@ double Simulator<Config>::ApproximateByFidelity(std::unique_ptr<dd::Package<Conf
...
-        if (!allLevels && remove * i > max_remove) {
-            max_remove      = remove * i;
-            nodes_to_remove = tmp;
+        if (!allLevels) {
+            if (i == 0) {
+                if (remove > max_remove) {
+                    max_remove      = remove;
+                    nodes_to_remove = tmp;
+                }
+            } else if (remove * i > max_remove) {
```
Comment (hillmich) [inline src/Simulator.cpp:187]: Good catch!

**Review by hillmich — COMMENTED** [review]
LGTM! Let's briefly discuss if we want to change a different name for `ApproximationMethod`.

File: include/CircuitSimulator.hpp (hunk 1 above)
Comment (burgholzer) [reply include/CircuitSimulator.hpp:15]: I like approximation strategy. Will adapt correspondingly.

Comment (burgholzer) [reply include/CircuitSimulator.hpp]: Done in https://github.com/cda-tum/ddsim/pull/186/commits/3ae652bcabcb016ae18e19b9e65a198140d6d0d0

**Review by hillmich — APPROVED** [review]
LGTM!

**Pull request merged**